### DISCUSSION TITLE
DKIM-Signature

### DISCUSSION THREAD

**Discussion by Nazar78** [berrystephenw/OctoText#126] [Ideas]
Can we have DKIM headers option added to outgoing emails? Signing the emails for personal domains helps lessen the chances they'll be flagged as spam.

**Comment by berrystephenw** [comment]
It looks like it is possible: https://russell.ballestrini.net/quickstart-to-dkim-sign-email-with-python/
But I'm not sure how I would test it. It's also going to require uploading a file (public key) and all the UI changes that would go along with that. It's something I could work on, but in all honesty my free time this summer is pretty booked up. 

I'm always willing to look at pull requests if you are interested in contributing.

**Comment by Nazar78** [comment]
@berrystephenw 

Thank you for replying. 

> But I'm not sure how I would test it.

Think [dkimpy](https://github.com/forwardemail/dkimpy) can be use to verify DKIM in raw email. I can also help with the live testing using my domain and mail server.

> It's also going to require uploading a file (public key)

It actually needs the private key and passphrase if required. Public key is placed as DNS TXT record. We can use strings as in a form of textarea for the private key and text input for the passphrase, domain and selector. These four are the minimum requirement for DKIM signing.

> I'm always willing to look at pull requests if you are interested in contributing.

I may be wrong but I noticed in the python dkim library it doesn't seems to support passphrase if the private key has one? Can you help to confirm? I've implemented DKIM in PHP and Perl but I'm not quite versed in Python.

Thank you.

**Comment by Nazar78** [comment]
I managed to sign the mail, although not in python because I'm not sure how to use passphrase in the dkim library, instead with postfix plus opendkim running on my Asus router. Still it's good to have the ability to sign emails in the plugin for those who doesn't wish to run a mail server.

![image](https://user-images.githubusercontent.com/2883688/167249157-8833b958-11e4-42a2-86f6-346d6e64e666.png)

Anyway setting this up I noticed OctoText will always start TLS by default. Another idea is to have an option to turn off STARTTLS for non-supported legacy smtp or relay servers.

**Comment by berrystephenw** [comment]
You can turn off TLS, but it will then use SSL. Are you suggesting that all encryption be turned off?

**Reply by Nazar78** [reply]
> You can turn off TLS, but it will then use SSL. Are you suggesting that all encryption be turned off?

Yes, an optiom to turn off all encryption to be used with legacy relay servers. Or simple replace that force SSL checkbox with a drop down menu containing encryption options: none, SSL or TLS.
